src/dll/edbgEcmdDll.C

```diff
@@ -1696,11 +1702,269 @@ uint32_t dllQueryHostMemInfoRanges( const std::vector<ecmdChipTarget> & i_target
 #endif // ECMD_REMOVE_MEMORY_FUNCTIONS
 
 #ifndef ECMD_REMOVE_INIT_FUNCTIONS
+
+#ifdef EDBG_ISTEP_CTRL_FUNCTIONS
+
+static edbgIPLTable g_edbgIPLTable;
+
+/**
+  * @brief Set the IPL mode
+  *
+  * @param enum ipl_mode i_mode - IPL mode (Interactive or Continous IPL mode)
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+uint32_t setIplMode(enum IplMode_t i_mode)
+{
+    uint32_t rc = ECMD_SUCCESS;
+    
+    /* If IPL mode is set to interactive then, SBE will be set in ISTEP mode
+     * and execute all SBE isteps in chip op mode */
+    if (i_mode == IPL_MODE_ISTEP)
+    {
+        ecmdChipTarget target;
+        constexpr uint32_t MBOX_SCRATCH_REG3 = 0x283a;
+        constexpr uint32_t MBOX_SCRATCH_REG8 = 0x283f;
+        ecmdDataBuffer dataBuffer;
+
+        target.chipTypeState = ECMD_TARGET_FIELD_VALID;
+        target.cageState = target.nodeState = target.slotState = target.posState = ECMD_TARGET_FIELD_VALID;
+        target.cage = target.node = target.slot = 0;
+
+        //This will be actually set by sbe_config_update istep procedure.
+        //For testing purpose we are setting it up.
+        dataBuffer.setWordLength(1);
+        dataBuffer.setWord(0, 0x86000000);
+        rc = dllPutCfamRegister(target, MBOX_SCRATCH_REG3, dataBuffer);
+        dataBuffer.setWord(0, 0x20000000);
+        rc = dllPutCfamRegister(target, MBOX_SCRATCH_REG8, dataBuffer);
+        if (rc)
+        {
+            out.error(EDBG_GENERAL_ERROR, FUNCNAME,
+                      "Failed dllPutCfamRegister()\n");
+            return rc;
+        }
+    }
+    return rc;
+}
+
+/**
+  * @brief This is a helper function to call the pdbg interface
+  *        to execute an istep
+  *
+  * @param uint16_t i_start_index - position of the istep
+  * @param uint16_t i_major - Major number to execute
+  * @param uint16_t i_minor_starrt - Minor number start
+  * @param uint16_t i_minor_starrt - Minor number end
+  *
+  * @return Upon success, ECMD_SUCCESS will be returned.  A reason code will
+  *         be returned if the execution fails.
+  */
+uint32_t dllIStepsHelper(uint16_t i_start_index, uint16_t i_major,
+                         uint16_t i_minor_start,
+                         uint16_t i_minor_end)
+{
+    struct pdbg_target* target;
+    uint32_t rc = ECMD_SUCCESS;
+    std::string output_buf_str;
+    std::string IStepName;
+
+    edbgIPLTable::edbgIStepDestination_t l_destination =
+                      edbgIPLTable::EDBG_ISTEP_INVALID_DESTINATION;
+    
+    // Iterate through each PIB class target
+    pdbg_for_each_class_target("pib", target)
```

Comment:
As per new plan BMC/SBE/HOST istep back-end function support is enabled as part of lib ipl except for istep power-on , No target specific check is required. 
I
Please note SBE istep should triggred only Master processor.

Reply:
please refer to the code here -> https://github.com/open-power/pdbg/blob/8b4611b5d8e7e2279fe4aa80c892fcfe10aa398d/src/istep.c . It is internally handled by sbe_istep() function.